orchestration/dagster_orchestration/hca_orchestration/solids/validate_egress.py

```diff
@@ -1,44 +1,29 @@
-from dagster import configured, DagsterType, InputDefinition, solid, String, StringSource, TypeCheckContext
+from dagster import configured, InputDefinition, solid, String, StringSource, TypeCheckContext
 from dagster.core.execution.context.compute import AbstractComputeExecutionContext
 
-from hca_manage.manage import HcaManage, ProblemCount
-from hca_orchestration.support.typing import DagsterConfigDict
+from hca_manage.manage import ProblemCount
+from hca_orchestration.support.hca_manage import hca_manage_from_solid_context
+from hca_orchestration.support.schemas import HCA_MANAGE_SCHEMA
+from hca_orchestration.support.typing import DagsterConfigDict, wrap_as_dagster_type
 
 
 def problem_count_typecheck(_: TypeCheckContext, value: object) -> bool:
     return isinstance(value, ProblemCount)
 
 
-DagsterProblemCount: DagsterType = DagsterType(
-    name="DagsterProblemCount",
-    type_check_fn=problem_count_typecheck,
+DagsterProblemCount = wrap_as_dagster_type(
+    ProblemCount,
     description="A simple named tuple to represent the different types of issues "
                 "present from the post process validation.",
 )
 
 
-POST_VALIDATION_SETTINGS_SCHEMA = {
-    "gcp_env": StringSource,
-    "dataset_name": String,
-}
-
-
-@solid(
-    required_resource_keys={"data_repo_client"},
-    config_schema={
-        **POST_VALIDATION_SETTINGS_SCHEMA,
-        "google_project_name": StringSource,
-    }
-)
+@solid(required_resource_keys={'data_repo_client'}, config_schema=HCA_MANAGE_SCHEMA)
 def base_post_import_validate(context: AbstractComputeExecutionContext) -> DagsterProblemCount:
     """
     Checks if the target dataset has any rows with duplicate IDs or null file references.
     """
-    validator = HcaManage(
-        environment=context.solid_config["gcp_env"],
-        project=context.solid_config["google_project_name"],
-        dataset=context.solid_config["dataset_name"],
-        data_repo_client=context.resources.data_repo_client)
+    validator = hca_manage_from_solid_context(context)
```

Comment:
waffling on whether this simplifies or obfuscates things, since it rests on the assumption that your solid is configured in a specific way that you can't perfectly guarantee.

my instinct is that HcaManage is better off as a resource, but that'll be a change we can make after raaid's refactor.

Reply:
I'd prefer to isolate solid level context + config to _just_ the solid, to avoid leaking that knowledge out of the toplevel "solids layer". We can just pass params to downstream method calls, which will help with testability and refactoring should we need to do so in the future.